statics/workflows/ping-mesh.yaml

```diff
@@ -0,0 +1,111 @@
+---
+UUID: "ce3c0765-c994-43ef-54f1-66abcb703101"
+name: "PingMesh(ICMP/TCP/UDP)"
+description: "Check Connectivity from Multiple Source to Single Destination"
+parameters:
+  - name: source
+    description: Source node
+    type: node
+  - name: destination
+    description: Destination node
+    type: node
+  - name: protocol
+    description: Protocol
+    type:  choice
+    default: ICMPv4
+    values:
+      - description: "Protocol : ICMPv4/Echo request"
+        value: ICMPv4
+      - description: "Protocol : TCP/IPv4"
+        value: TCP
+      - description: "Protocol : UDP/IPv4"
+        value: UDP
+source: |
+  function PingMesh(from, to, protocol) {
+    var dict = {};
+    var From = {};
+    var sources = [];
+    sources.push(from);
+    var i = 1;
+    var capture = new Capture();
+    capture.GremlinQuery = "G.V().Has('TID', '" + to + "')";
+    var packetInjection = new PacketInjection();
+    if (protocol == "ICMPv4") {
+      var random = Math.floor(Math.random() * 65535);
+    } else {
+      var random = 1024 + Math.floor(Math.random() * (65535-1024));
+    }
+    return client.captures.create(capture).then(function (c) {
+      capture = c
+    }).then(function () {
+      sources.forEach(function(source) {
+        packetInjection.Src = "G.V().Has('TID', '" + source + "')"
+        packetInjection.Dst = "G.V().Has('TID', '" + to + "')"
+        packetInjection.Count = 5
+        return client.G.V().Has("TID", from).then(
+          function (nodes) {
+          if (nodes[0].Metadata.Neutron && nodes[0].Metadata.Neutron.IPV4) {
+            packetInjection.SrcIP = nodes[0].Metadata.Neutron.IPV4[0]
+          }
+          if (nodes[0].Metadata.ExtID && nodes[0].Metadata.ExtID["attached-mac"]) {
+            packetInjection.SrcMAC = nodes[0].Metadata.ExtID["attached-mac"]
+          }
+          if (protocol == "ICMPv4") {
+            packetInjection.Type = "icmp4"
+            packetInjection.ICMPID = random + i + Math.floor(Math.random() * 1000);
+            i +=  Math.floor(Math.random() * 1000);
+            From[source] = packetInjection.ICMPID;
+          }
+          if (protocol == "TCP") {
+            packetInjection.Type = "tcp4"
+            packetInjection.SrcPort = random + i;
```

Comment:
I used a separate var because for random function may be the probability to get the same src and dst port no. (by 2 diff source) is tends to zero but it is not zero.
Yes but i can do this with the information of the source node like; IP and MAC; in gremlin query to get the exact flows on the destination node (coming from the source node).

Thanks!